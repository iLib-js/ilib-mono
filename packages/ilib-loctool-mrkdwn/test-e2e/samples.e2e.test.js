/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-mrkdwn samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require("path");
const fs = require("fs");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("ilib-internal");

describe("samples", () => {
    describe("slack", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "slack");
        const xliffPath = path.resolve(projectPath, "sample-slack-json-extracted.xliff");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            if (fs.existsSync(xliffPath)) {
                fs.unlinkSync(xliffPath);
            }
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
