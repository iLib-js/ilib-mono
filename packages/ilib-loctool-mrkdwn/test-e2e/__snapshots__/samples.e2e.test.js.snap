// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples slack should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="translations/strings.js" source-language="en-US" product-name="sample-slack-json">
    <body>
      <trans-unit id="1" resname="string1" restype="string" datatype="mrkdwn">
        <source>Hello</source>
        <note> comment for string1</note>
      </trans-unit>
      <trans-unit id="2" resname="string2" restype="string" datatype="mrkdwn">
        <source>Thank &lt;c0&gt;you&lt;/c0&gt;</source>
        <note> comment for string2</note>
      </trans-unit>
      <trans-unit id="3" resname="string3" restype="string" datatype="mrkdwn">
        <source>This &lt;c0&gt;is a &lt;c1&gt;test&lt;/c1&gt; with a unique id&lt;/c0&gt;</source>
        <note> another comment, this time for string3</note>
      </trans-unit>
      <trans-unit id="4" resname="string4" restype="string" datatype="mrkdwn">
        <source>This is a &lt;c0&gt;test&lt;/c0&gt; with a slack &lt;c1&gt;link&lt;/c1&gt; in it and a &lt;c2&gt;User&lt;/c2&gt; reference.</source>
        <note> string4 is the target of this comment</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
